bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/mapping/VolumioData.java

```diff
@@ -0,0 +1,357 @@
+    private boolean repeatDirty;
+
+    private boolean random = false;
+    private boolean randomDirty;
+
+    public void update(JSONObject jsonObject) throws JSONException {
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_TITLE)) {
+            setTitle(jsonObject.getString(VolumioBindingConstants.CHANNEL_TITLE));
+        } else {
+            setTitle("");
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_ALBUM)
+                && !jsonObject.isNull(VolumioBindingConstants.CHANNEL_ALBUM)) {
+            setAlbum(jsonObject.getString(VolumioBindingConstants.CHANNEL_ALBUM));
+        } else {
+            setAlbum("");
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_VOLUME)) {
+            setVolume(jsonObject.getInt(VolumioBindingConstants.CHANNEL_VOLUME));
+        } else {
+            setVolume(0);
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_ARTIST)) {
+            setArtist(jsonObject.getString(VolumioBindingConstants.CHANNEL_ARTIST));
+        } else {
+            setArtist("");
+        }
+
+        /* Special */
+        if (jsonObject.has("status")) {
+            setState(jsonObject.getString("status"));
+        } else {
+            setState("pause");
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_TRACK_TYPE)) {
+            setTrackType(jsonObject.getString(VolumioBindingConstants.CHANNEL_TRACK_TYPE));
+        } else {
+            setTrackType("");
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_COVER_ART)
+                && !jsonObject.isNull(VolumioBindingConstants.CHANNEL_COVER_ART)) {
+            setCoverArt(jsonObject.getString(VolumioBindingConstants.CHANNEL_COVER_ART));
+        } else {
+            setCoverArt(null);
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_PLAY_RANDOM)
+                && !jsonObject.isNull(VolumioBindingConstants.CHANNEL_PLAY_RANDOM)) {
+            setRandom(jsonObject.getBoolean(VolumioBindingConstants.CHANNEL_PLAY_RANDOM));
+        } else {
+            setRandom(false);
+        }
+
+        if (jsonObject.has(VolumioBindingConstants.CHANNEL_PLAY_REPEAT)
+                && !jsonObject.isNull(VolumioBindingConstants.CHANNEL_PLAY_REPEAT)) {
+            setRepeat(jsonObject.getBoolean(VolumioBindingConstants.CHANNEL_PLAY_REPEAT));
+        } else {
+            setRepeat(false);
+        }
+    }
+
+    public StringType getTitle() {
+        return new StringType(title);
+    }
+
+    public void setTitle(String title) {
+        if (!title.equals(this.title)) {
+            this.title = title;
+            this.titleDirty = true;
+        } else {
+            this.titleDirty = false;
+        }
+    }
+
+    public StringType getAlbum() {
+        return new StringType(album);
+    }
+
+    public void setAlbum(String album) {
+        if ("null".equals(album)) {
+            album = "";
```

Comment:
Assigning this method param seems like a bug:

```suggestion
            this.album = "";
```

Reply:
solved